### ISSUE TITLE
SlickGoTo doesn't seem to work in componentDidUpdate

### ISSUE THREAD

**Issue by kevindavis** [akiran/react-slick#440]
When passing some new parameters to the component containing the slider, I'd like to have the Slider start at a given slide. The SlickGoTo example shows how to jump to a given slide based on user input. I tried to use the componentDidUpdate callback without success: 

`  
componentDidUpdate: function(){
    this.refs.slider.slickGoTo(this.props.intro_slide);
}
`

My understanding of the initialSlide property is that it only applies to the first slide displayed when the component mounts. 

Is there another way to accomplish this?

**Comment by kevindavis** [comment]
fwiw the deprecated slickGoTo property kind of works for this purpose (though I'd really prefer it didn't animate flipping through the slides)

**Comment by akiran** [comment]
Can you provide jsfiddle replicating you usecase.

**Comment by akiran** [comment]
componentDidUpdate is not the right place to call slickGoTo method because it trigger state change and infinite number of re-renders

Ideally slickGoTo should be called in event handlers like onClick

**Comment by kevindavis** [comment]
Camping this weekend but will try to make a minimal example when I get back
to a computer. Basically I'm passing parameters a couple of times with
different data to render. Different numbers of slides and different start
positions.

I'm pretty new to React so there might be a better way of doing this..

Thanks for being so responsive!

-- 
Kevin Davis

**Comment by kevindavis** [comment]
Here's an attempt to describe my scenario distilled to the smallest way I can think of. The 'change deck' button is a contrived way of changing the state on the parent component. The real components are more complicated etc..

https://jsfiddle.net/kevindavis/o1jbs86x/15/

Really appreciate your help

**Comment by kevindavis** [comment]
Ideas on how I should approach this?

**Comment by Josh-ES** [comment]
Not sure of your exact use case from the fiddle, but what you've said suggests you should use the 'componentWillReceiveProps' method on your component: https://facebook.github.io/react/docs/component-specs.html#updating-componentwillreceiveprops. You can then call the 'slickGoTo' method within that. Let me know if that doesn't answer your question!

```js
componentWillReceiveProps: function(nextProps) {
  if (nextProps.start !== this.props.start) {
    this.refs.slider.slickGoTo(nextProps.start);
  }
}
```

**Comment by kevindavis** [comment]
Oh man. This has been really confusing me. The example you gave does work - thanks @Josh-ES !

However, when I tried to replicate this change in my app it didn't work.. leaving me totally stumped. It does seems to work when the start slide you pass in is 2 but doesn't when the start slide is 0 (issue #441). Further investigating, it seems to be even more complicated -- there are several cases where the start slide is not 0 and it still doesn't go where you'd expect. @Kaveckas PR (#450) fixes the issue for me, but it looks like there might be something else going on.

Here's an updated example that doesn't work:
https://jsfiddle.net/kevindavis/o1jbs86x/27/

Here's a sheet of cases specific to my example:
https://docs.google.com/spreadsheets/d/1zzQqNJUAR6fEZtqZZ8qVGeJJpG5bq_cfoSbecp6ulDQ/edit?usp=sharing

:-\

Thanks for everyone's help!

**Comment by akiran** [comment]
@kevindavis Can you explain the expected functionality of https://jsfiddle.net/kevindavis/o1jbs86x/27/

When I first click on the button, slickGoTo is working as expected.
Subsequent clicks are not having effect because start state is already set to 2 hence not triggering slickGoTo again

**Comment by kevindavis** [comment]
Sorry, I could have been a lot more clear. The fix for #441 solves a lot of these cases (described in the spreadsheet) but here is an example of one that isn't fixed:

1. go to https://jsfiddle.net/kevindavis/o1jbs86x/30/
2. advance the slides by one to see 'slide2' 
3. click the 'change deck' button

expected behavior: to see "slide-n1" (start position 1 in the second deck)
actual behavior: see "slide-n2"

**Comment by akiran** [comment]
Yes,  What @Josh-ES said about if clause if correct.

I merged #450 and made a release.
